### DISCUSSION TITLE
oop_use_cases

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/labview_book#21] [Announcements]
# 类的应用实例

https://lv.qizhen.xyz/oop_use_cases

**Comment by Arita9527** [comment]
“再创建一个数据为空的基本类做为它们共同的父类”
可以先创建子类，再创建父吗，那怎么选择他们的继承呢

**Reply by ruanqizhen** [reply]
可以，但是类的继承一定要在子类的设置对话框中进行设置：
![](https://github.com/ruanqizhen/labview_book/raw/main/docs/images_2/image76.png)

**Comment by Raveeeen** [comment]
<picture>
  <img src="https://lv.qizhen.xyz/assets/images/z032-02972284077e26e5ab801201c3407cf6.png">
</picture>

你好
想請教一下圖片中右側輸出的next是甚麼數據類型? 是iterator類嗎?
因為我看文中描述提到 "得到下个节点的引用之后，再包装成一个 Iterator 的对象。"
所以看起來next像是一個iterator對象, 但是他的wire的風格似乎和DoublyNodeRef In/Out有點不太一樣 這對輸入輸出從文中描述來看應該也是iterator對象.
再請您幫忙解答 
謝謝

**Reply by ruanqizhen** [reply]
是的，输出 next 类型与上面的两个 in/out 类型是一样的，是 iterator 对象。上面两个控件如果命名为 Iterator in/out 会更合理，我更新了贴图。
它们的数据类型相同，数据线的风格一般情况下应当是一样的。但是在 lvclass 的 VI 中，如果输入输出 class 控件被设置成了“Dynamic Dispatch”，也就是说它们是用来代表 self 自身这个对象的话，它们的数据线就会被加粗。看起来和在其它 VI 中使用时就会稍有不同。

**Comment by wyxfhb** [comment]
阮老师，你好。想请教一下，简单列表（List）数据结构章节中，Demo实例里面，添加list表的节点是通过循环和移位寄存器实现的，通过循环次数确定添加的节点数。这里面有个问题不太明白，印象中只有数组和集合可以通过这种方式添加数组元素或者集合的元素，类的一个对象为什么具有这种属性呢？类的一个对象，通过这种方式容纳多个对象，这个机制实现的原理是什么呢？

**Reply by ruanqizhen** [reply]
这要看类是怎么设计的。循环是需要每个迭代都做类似的工作。演示中的这个类，每个示例中还包含了同样类的另一个实例，这就形成了一种嵌套关系，每一层的结构都和其它层是类似的，都可以使用同样的操作。（参考演示程序上面的示例图。）因此，它可以通过循环，每次迭代都进入到内部一层，构建，或者读取数据。

**Comment by lixin200111** [comment]
阮老师，请问后续有考虑增加操作者框架的内容吗

**Reply by ruanqizhen** [reply]
我是考虑添加这部分内容的，不过可能还要再等一段时间。我还在整理前面的基础内容，从这本书的阅读统计来看，大多数浏览记录都集中在前面几章基础内容里。等把基础内容整理差不多了，再着手写这些复杂的框架。

**Reply by lixin200111** [reply]
很期待阮老师的讲解，操作者框架的资料比较少而且基本全英文，本身还比较复杂

**Reply by Cliso** [reply]
期待+1，
希望有AF中接口的详细讲解。NIexample里的范例程序的接口用于多操作者的部分，讲解就几句话，不是很好理解。
